### PR TITLE
Add Repository Security Advisories APIs

### PR DESCRIPTION
**Pull request by anishrajan25**
Relates to #2866 

Integrates the following 2 APIs:

- [Lists repository security advisories for an organization - /orgs/{org}/security-advisories](https://docs.github.com/en/rest/security-advisories/repository-advisories?apiVersion=2022-11-28#list-repository-security-advisories-for-an-organization)
- [Lists security advisories in a repository - /repos/{owner}/{repo}/security-advisories](https://docs.github.com/en/rest/security-advisories/repository-advisories?apiVersion=2022-11-28#list-repository-security-advisories)

### CODE REVIEW

**Comment by anishrajan25** [comment]
@gmlewis The PR is ready for review. Could you please have a look as per your availability?

**Review by gmlewis — CHANGES_REQUESTED** [review]
I don't have time for a full review right now, but at a quick glance, it looks like you are defining a whole bunch of new structs... possibly more than necessary.  We try to reuse structs when the meaning is similar if possible.

It also looks like you are redefining `ListOptions` and/or `ListCursorOptions` when they could be embedded into other options... please look around this repo for examples on how to do this.

Third, all new structs must have complete sentences ending in a period for the auto-generated godocs to be formatted correctly.  Please fix this.

Additionally, it looks like there are linting and/or test errors that need fixing. Please address.

**Comment by anishrajan25** [comment]
@gmlewis I have analysed the existing structs and prepared a document capturing the possibilities of embedding some structs to prevent redefining some fields.

Could you please have a look at [this document](https://docs.google.com/document/d/10ezC0VHeUWk3WBNZJjOlOld97XOLeRQpNA34Xsz0BFM/edit?usp=sharing) and add comments wherever necessary to provide your inputs?

I will fix the comments to end with period, linting, and test errors in the next commit along with updating the struct definitions.

**Comment by gmlewis** [comment]
I'll be back on 9/18 when I can take a look at this. Sorry for the delay and inconvenience.

**Comment by anishrajan25** [comment]
No worries. You can have a look whenever you are available ✌🏼

**Comment by gmlewis** [comment]
@anishrajan25 - I'm back, but none of the comments I left above:
https://github.com/google/go-github/pull/2902#pullrequestreview-1599004532
have been addressed.  Please fix these issues before we can continue.

**Comment by anishrajan25** [comment]
Hi @gmlewis , apologies for the delay in response.

I have created [this document](https://docs.google.com/document/d/10ezC0VHeUWk3WBNZJjOlOld97XOLeRQpNA34Xsz0BFM/edit?usp=sharing), capturing the possibilities of embedding some structs to prevent redefining some fields. 
Could you please have a look and add comments wherever necessary to provide your input?

**Comment by anishrajan25** [comment]
I have updated the existing structs and reused them instead of creating new ones wherever possible. I have fixed the comments as well. @gmlewis Could you please have a look?

File: github/security_advisories.go [hunk 1]
```diff
@@ -12,6 +12,50 @@ import (
...
+// Permissions represent a team's permissions.
+type Permissions struct {
+	Admin    *bool `json:"admin,omitempty"`
+	Pull     *bool `json:"pull,omitempty"`
+	Push     *bool `json:"push,omitempty"`
+	Triage   *bool `json:"triage,omitempty"`
+	Maintain *bool `json:"maintain,omitempty"`
+}
+
+// ListRepositorySecurityAdvisoriesOptions specifies the optional parameters to lists the repository security advisories.
```
Comment (gmlewis) [inline github/security_advisories.go:43]:
```suggestion
// ListRepositorySecurityAdvisoriesOptions specifies the optional parameters to list the repository security advisories.
```

File: github/security_advisories.go [hunk 2]
```diff
@@ -35,3 +79,51 @@ func (s *SecurityAdvisoriesService) RequestCVE(ctx context.Context, owner, repo,
 
 	return resp, nil
 }
+
+// ListRepositorySecurityAdvisoriesForOrg lists the repository security advisories for an organization.
+//
+// Github API docs: https://docs.github.com/en/rest/security-advisories/repository-advisories?apiVersion=2022-11-28#list-repository-security-advisories-for-an-organization
```
Comment (gmlewis) [inline github/security_advisories.go:85]:
```suggestion
// GitHub API docs: https://docs.github.com/en/rest/security-advisories/repository-advisories?apiVersion=2022-11-28#list-repository-security-advisories-for-an-organization
```

File: github/security_advisories.go [hunk 3]
```diff
@@ -35,3 +79,51 @@ func (s *SecurityAdvisoriesService) RequestCVE(ctx context.Context, owner, repo,
...
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return advisories, resp, nil
+}
+
+// ListRepositorySecurityAdvisories lists the security advisories in a repository.
+//
+// Github API docs: https://docs.github.com/en/enterprise-cloud@latest/rest/security-advisories/repository-advisories?apiVersion=2022-11-28#list-repository-security-advisories
```
Comment (gmlewis) [inline github/security_advisories.go:109]:
```suggestion
// GitHub API docs: https://docs.github.com/en/enterprise-cloud@latest/rest/security-advisories/repository-advisories?apiVersion=2022-11-28#list-repository-security-advisories
```

File: github/security_advisories.go [hunk 4]
```diff
@@ -35,3 +79,51 @@ func (s *SecurityAdvisoriesService) RequestCVE(ctx context.Context, owner, repo,
...
+func (s *SecurityAdvisoriesService) ListRepositorySecurityAdvisories(ctx context.Context, owner string, repo string, opt *ListRepositorySecurityAdvisoriesOptions) ([]*SecurityAdvisory, *Response, error) {
```
Comment (gmlewis) [inline github/security_advisories.go:110]:
```suggestion
func (s *SecurityAdvisoriesService) ListRepositorySecurityAdvisories(ctx context.Context, owner, repo string, opt *ListRepositorySecurityAdvisoriesOptions) ([]*SecurityAdvisory, *Response, error) {
```

**Review by gmlewis — CHANGES_REQUESTED** [review]

File: github/security_advisories_test.go [hunk 5]
```diff
@@ -50,3 +53,265 @@ func TestSecurityAdvisoriesService_RequestCVE(t *testing.T) {
...
+
+		w.WriteHeader(http.StatusOK)
+		w.Write([]byte(`[{"ghsa_id": 12334354}]`))
+	})
+
+	ctx := context.Background()
+	advisories, resp, err := client.SecurityAdvisories.ListRepositorySecurityAdvisories(ctx, "o", "r", nil)
+	if err == nil {
+		t.Errorf("Expected unmarshal error")
+	} else if !strings.Contains(err.Error(), "json: cannot unmarshal number into Go struct field RepoSecurityAdvisory.ghsa_id of type string") {
```
Comment (gmlewis) [inline github/security_advisories_test.go:256]:
```suggestion
	} else if !strings.Contains(err.Error(), "json: cannot unmarshal number into Go struct field SecurityAdvisory.ghsa_id of type string") {
```

**Review by gmlewis — CHANGES_REQUESTED** [review]

**Comment by gmlewis** [comment]
Please check that test pass locally with `go test ./...` before pushing.

**Comment by anishrajan25** [comment]
There are some failing test cases. I will update once the test cases are fixed.

**Comment by anishrajan25** [comment]
@gmlewis I have fixed the test cases as well. All the tests are passing on my local

**Comment by gmlewis** [comment]
Please run `go generate ./...` as described in CONTRIBUTING.md and push the results to this PR.

**Review by gmlewis — APPROVED** [review]
Thank you, @anishrajan25 !
LGTM.

Awaiting second LGTM+Approval from any other contributor to the repo before merging.

**Review by valbeat — APPROVED** [review]

**Comment by gmlewis** [comment]
Thank you, @valbeat !
Merging.

**Pull request merged**